service/mantle/shipstation/OmsShipmentService.xml

```diff
@@ -0,0 +1,395 @@
+            <parameter name="shipmentId"/>
+            <parameter name="shipmentPackageSeqId"/>
+        </out-parameters>
+        <actions>
+            <entity-find-one entity-name="mantle.order.OrderHeader" value-field="orderHeader"/>
+            <entity-find-one entity-name="mantle.order.OrderPart" value-field="orderPart"/>
+            <if condition="!(orderPart.statusId in ['OrderApproved', 'OrderSent'] || orderHeader.statusId in ['OrderApproved', 'OrderSent'])">
+                <return type="danger" message="Cannot create Shipment for Order Part in status ${orderPart.statusId}"/>
+            </if>
+
+            <!-- does the OrderPart have shippable items? -->
+            <!--<entity-find-related value-field="orderPart" relationship-name="mantle.order.OrderItem" list="orderItemList"
+                                 order-by-list="['orderItemSeqId']"/>-->
+            <!--<if condition="!orderItemList.find({ it.product?.productTypeEnumId in ['PtAsset', 'PtDigitalAsset', 'PtAssetUse', 'PtPickAssembly'] })">
+                <return type="warning"
+                        message="Order ${orderId} part ${orderPartSeqId} has no shippable product items (type asset or asset use)"/>
+            </if>-->
+
+            <!-- and not already on a Shipment? -->
+            <set field="totalQtyToShip" from="0.0"/>
+            <set field="activeShipmentIds" from="new TreeSet()"/>
+            <entity-find entity-name="moqui.basic.EnumGroupMember" list="productItemTypeEgms" cache="true">
+                <econdition field-name="enumGroupEnumId" value="EngItemsProduct"/>
+            </entity-find>
+            <set field="productItemTypes" from="productItemTypeEgms*.enumId"/>
+
+            <iterate list="orderItemList" entry="orderItem">
+                <entity-find-one entity-name="mantle.order.OrderItem" value-field="Item">
+                    <field-map field-name="orderId" from="orderId"/>
+                    <field-map field-name="orderItemSeqId" from="orderItem.orderItemSeqId"/>
+                </entity-find-one>
+
+                <!-- if no productId or not a product item type skip it -->
+                <if condition="!Item.productId || !productItemTypes.contains(Item.itemTypeEnumId)">
+                    <continue/>
+                </if>
+
+                <!-- see if Product is a physical item -->
+                <entity-find-one entity-name="mantle.product.Product" value-field="product" cache="true">
+                    <field-map field-name="productId" from="Item.productId"/>
+                </entity-find-one>
+                <if condition="!(product.productTypeEnumId in ['PtAsset', 'PtDigitalAsset', 'PtAssetUse', 'PtPickAssembly'])">
+                    <continue/>
+                </if>
+
+                <!-- determine quantityNotShipped -->
+                <set field="quantityNotShipped"
+                     from="(Item.quantity != null ? Item.quantity : 1.0) * (Item.selectedAmount ?: 1.0)"/>
```

Comment:
Why are we taking quantityNotShipped from order item? Shouldn't we be checking quantity from the incoming items list?

Reply:
@gurveenbagga maam, quantityNotShipped is a variable that gives us the value of Item quantity that is not shipped till now, and there we compare our quantity with this value.

Reply:
I think we should check on the quantity of the individual item, if it is already shipped, log the warning and continue the items loop.